core/src/main/java/hudson/model/Computer.java

```diff
@@ -669,41 +672,59 @@ public boolean isLaunchSupported() {
     @Exported
     @Deprecated
     public boolean isTemporarilyOffline() {
-        return temporarilyOffline;
+        var node = getNode();
+        return node != null && node.isTemporarilyOffline();
     }
 
     /**
      * @deprecated as of 1.320.
-     *      Use {@link #setTemporarilyOffline(boolean, OfflineCause)}
+     *      Use {@link #setTemporaryOfflineCause(OfflineCause)}
      */
     @Deprecated
     public void setTemporarilyOffline(boolean temporarilyOffline) {
-        setTemporarilyOffline(temporarilyOffline, null);
+        setTemporaryOfflineCause(temporarilyOffline ? new OfflineCause.LegacyOfflineCause() : null);
+    }
+
+    /**
+     * @deprecated as of TODO.
+     *      Use {@link #setTemporaryOfflineCause(OfflineCause)} instead.
+     */
+    @Deprecated
+    public void setTemporarilyOffline(boolean temporarilyOffline, OfflineCause cause) {
```

Comment:
The javadoc is lost here especially that the second argument is only considered when the first is true. But maybe this doesn't matter as the method is now set to deprecated.

Reply:
Yes, I'm dropping existing javadoc to avoid repeats, and only refer to the current methods.